src/core/public/core_system.ts

```diff
@@ -44,6 +46,14 @@ export class CoreSystem {
       injectedMetadata,
     });
 
+    this.fatalErrors = new FatalErrorsService({
+      rootDomElement,
+      injectedMetadata: this.injectedMetadata,
+      stopCoreSystem: () => {
```

Comment:
Your argument for `onFirstError` is very tempting, but I think this situation is a little different than normal because the `FatalErrorService` actually requires specific actions to be completed before it acts on the first error and takes full control of the `rootDomElement`.

Normally I would suggest using something like `fatalErrors.get$().pipe(first()).toPromise()` to "register a callback for the first error", but if we did this then the core system would actually be shutdown after the `FatalErrorService` has already overwritten the entire contents of the `rootDomElement`. This could *probably* be fixed by not using `toPromise()`, or by subscribing on an async scheduler within the `FatalErrorService`, but then we have a weird implicit dependency based on fragile timing.

We could also just follow your specific `onFirstError()` param suggestion, which doesn't have this timing issue, but then I think the fact that the `FatalErrorService` relies on everything unmounting from the `rootDomElement` is less clearly expressed. Requiring a `stopCoreSystem()` param expresses this a bit better I think... Thoughts?